server/src/main/java/com/genymobile/scrcpy/Intents.java

```diff
@@ -0,0 +1,42 @@
+package com.genymobile.scrcpy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.ListIterator;
+
+public class Intents {
+
+    public static void intentAnnounce(String action,
+                                        String uriContent,
+                                        HashMap<String, String> extras ){
+
+        ArrayList<String> list = new ArrayList<>(Arrays.asList(
+                "am", "broadcast",
+                "-a", "com.genymobile.scrcpy." + action,
+                "-d", "scrcpy-status:" + uriContent,
+                "--ez", "STARTUP", extras.get("STARTUP"),
+                "--ez", "SHUTDOWN", extras.get("SHUTDOWN")
+        ));
+
+        ListIterator<String> iter = list.listIterator(list.size());
+        for (String e = iter.previous(); iter.hasPrevious(); e = iter.previous()) {
+            if(e == null){
+                // If there's no value, remove the entry
+                iter.remove();
+                iter.previous();
+                iter.remove();
+                iter.previous();
+                iter.remove();
+            }
+        }
```

Comment:
I thought it's easier to read if just adding all and then remove the unused ones than to just add the ones provided.
Doing the other way is fine too